cmake/std/atdm/van1-tx2/environment.sh

```diff
@@ -0,0 +1,153 @@
+################################################################################
+#
+# Set up env on a ARM ATSE builds of Trilinos
+# This source script gets the settings from the ATDM_CONFIG_BUILD_NAME var.
+#
+################################################################################
+
+echo "Using ARM ATSE compiler stack $ATDM_CONFIG_COMPILER to build $ATDM_CONFIG_BUILD_TYPE code with Kokkos node type $ATDM_CONFIG_NODE_TYPE"
+
+#
+# Assert and set KOKKOS_ARCH
+#
+
+if   [[ "$ATDM_CONFIG_KOKKOS_ARCH" == "DEFAULT" ]] \
+  || [[ "$ATDM_CONFIG_KOKKOS_ARCH" == "" ]] \
+  ; then
+  export ATDM_CONFIG_KOKKOS_ARCH=ARMv8-TX2
+else
+  echo
+  echo "***"
+  echo "*** ERROR: Only one arch is supported this system!  Remove any arch keywords from build name '${ATDM_CONFIG_BUILD_NAME}'"
+  echo "***"
+  return
+fi
+
+export ATDM_CONFIG_ENABLE_SPARC_SETTINGS=ON
+export ATDM_CONFIG_USE_NINJA=ON
+
+export ATDM_CONFIG_BUILD_COUNT=40  # Assume building on a compute node!
+
+if [[ "$ATDM_CONFIG_NODE_TYPE" == "OPENMP" ]] ; then
+  export ATDM_CONFIG_CTEST_PARALLEL_LEVEL=16
+  export OMP_NUM_THREADS=2
+else
+  export ATDM_CONFIG_CTEST_PARALLEL_LEVEL=32
+  export OMP_PROC_BIND=FALSE
+  export OMP_NUM_THREADS=1
+fi
+
+#
+# Load the modules
+#
+
+module purge
+
+if [[ "$ATDM_CONFIG_COMPILER" == "ARM-20.0_OPENMPI-4.0.2" ]]; then
+  module load devpack-arm
+  module unload yaml-cpp
+  # provides numpy module for empire
+  module load python/3.6.8
+  module load arm/20.0
+  # Check if openmpi is already loaded. If it is, swap it. Otherwise, just load ompi4.
+  if [[ ! -z $(module list openmpi | grep '1)' | awk -F ' ' '{print $2}') ]]; then
+    module swap $(module list openmpi | grep '1)' | awk -F ' ' '{print $2}') openmpi4/4.0.2
+  else
+    module load openmpi4/4.0.2
+  fi
+  module load armpl/20.0.0
+
+  export LAPACK_ROOT="$ARMPL_LIB"
+  export ATDM_CONFIG_LAPACK_LIBS="-L${LAPACK_ROOT};-larmpl_ilp64_mp"
+  export ATDM_CONFIG_BLAS_LIBS="-L${LAPACK_ROOT};-larmpl_ilp64_mp"
+else
+  echo
+  echo "***"
+  echo "*** ERROR: COMPILER=$ATDM_CONFIG_COMPILER is not supported!"
+  echo "***"
+  return
+fi
+
+# Common modules for all builds
+module load ninja
```

Comment:
@bartlettroscoe: Should this be `module load ninja/1.8.2` to guard against something changing under the hood going forward?

Reply:
That is worth considering.  On the other hand, if we get an updated ninja automatically, I would not complain either.  I think it is important that we try to be testing the default envs on these machines since I suspect that is what users depend on.  Really we need to be using what the ATDM APPs are using so really this should be their choice.   How about we ask them?  Do they want us to be testing with the most recent Fortran-capable ninja on that system or do they want us to be testing against a fixed version of ninja?